### PR TITLE
buildkite_insights: Add script for finding segfaults

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/buildkite_insights/util/buildkite_api.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def get(
         if isinstance(result, dict) and result.get("message"):
             raise RuntimeError(f"Something went wrong! ({result['message']})")
 
-        params["created_to"] = result[-1]["created_at"]
+        params["page"] = str(int(params.get("page", "1")) + 1)
```
Comment (def-) [inline misc/python/materialize/buildkite_insights/util/buildkite_api.py:49]: This seems to work in all paginated results in Buildkite while the created_to  is only possible in some specific API endpoints.

File: misc/python/materialize/buildkite_insights/segfaults/find.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import time
+
+from materialize.buildkite_insights.util import buildkite_api
+
+
+def main() -> None:
```
Comment (nrainer-materialize) [inline misc/python/materialize/buildkite_insights/segfaults/find.py:17]: Optional: Consider using `fetch_builds` and iterating over the possibly three relevant pipelines.

File: misc/python/materialize/buildkite_insights/segfaults/find.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    for build in data:
+        request_path = f"organizations/materialize/pipelines/{build['pipeline']['slug']}/builds/{build['number']}/artifacts"
+        params = {"per_page": "100"}
+        result = buildkite_api.get(request_path, params, max_fetches=None)
+        for artifact in result:
+            if (
+                "core" in artifact["filename"]
+                and build["pipeline"]["slug"] != "coverage"
+                and artifact["file_size"] > 100000
```
Comment (nrainer-materialize) [inline misc/python/materialize/buildkite_insights/segfaults/find.py:34]: Can you add an explanatory comment to this line?

File: misc/python/materialize/buildkite_insights/segfaults/find.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            ):
+                print(
+                    f"{build['started_at']}: {artifact['filename']} in https://buildkite.com/materialize/{build['pipeline']['slug']}/builds/{build['number']}#{artifact['job_id']}"
+                )
+        time.sleep(2)
```
Comment (nrainer-materialize) [inline misc/python/materialize/buildkite_insights/segfaults/find.py:39]: Needed or temporary for testing?

**Review by nrainer-materialize — APPROVED** [review]

File: misc/python/materialize/buildkite_insights/segfaults/find.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+from materialize.buildkite_insights.util import buildkite_api
+
+
+def main() -> None:
+    # Used to find recent instances of https://github.com/MaterializeInc/materialize/issues/24644
+    # 2 weeks ~ 2000 builds
+    data = buildkite_api.get(
+        "organizations/materialize/builds",
+        {"include_retried_jobs": "true", "per_page": "100"},
+        max_fetches=20,
```
[the remark this answers is not in the document]
Comment (def-) [reply misc/python/materialize/buildkite_insights/segfaults/find.py:23]: It is! But I couldn't think of a better way. Without the sleep we ran into buildkite ratelimits even.

File: misc/python/materialize/buildkite_insights/segfaults/find.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import time
+
+from materialize.buildkite_insights.util import buildkite_api
+
+
+def main() -> None:
+    # Used to find recent instances of https://github.com/MaterializeInc/materialize/issues/24644
+    # 2 weeks ~ 2000 builds
+    data = buildkite_api.get(
```
[the remark this answers is not in the document]
Comment (def-) [reply misc/python/materialize/buildkite_insights/segfaults/find.py:20]: I'm not confident that this segfault will always cause a test failure (but so far it did)

File: misc/python/materialize/buildkite_insights/segfaults/find.py (hunk 4 above)
Comment (def-) [reply misc/python/materialize/buildkite_insights/segfaults/find.py:39]: I just wanted to be nice to buildkite, also hit their ratelimit at some point.

**Pull request merged**